GitCommands/CommitDataManager.cs

```diff
@@ -58,15 +58,12 @@ public void UpdateBody(CommitData commitData, bool appendNotesOnly, out string?
             {
                 if (!string.IsNullOrWhiteSpace(data))
                 {
-                    commitData.Body += $"\nNotes:\n    {GetModule().ReEncodeCommitMessage(data.Replace('\v', '\n'))}";
+                    commitData.Notes = GetModule().ReEncodeCommitMessage(data.Replace('\v', '\n'));
                 }
             }
             else
             {
-                string[] lines = data.Split(Delimiters.LineAndVerticalFeed);
-
-                // Commit message is not re-encoded by Git when format is given
-                commitData.Body = GetModule().ReEncodeCommitMessage(ProcessDiffNotes(startIndex: 0, lines));
+                (commitData.Body, commitData.Notes) = ProcessDiffNotes(data.Split(Delimiters.LineAndVerticalFeed)); // Commit message is not re-encoded by Git when format is given
```

Comment:
I do not get the comment. `GetModule().ReEncodeCommitMessage` is called by the nested function, which should be kept `static`.
```suggestion
                (commitData.Body, commitData.Notes) = ProcessDiffNotes(data.Split(Delimiters.LineAndVerticalFeed), GetModule().ReEncodeCommitMessage);
```

Reply:
Comment location - ok, should have been moved along with ReEncode... calls. Whatever it means.

Nested functions.. I don`t see any nested (local?) functions around and don`t get what it has to do with anything. I understand, that in some scenarios local funcions should better be static, but that does not apply here.

Reply:
Yes, `ProcessDiffNotes` is not nested, but it should not be turned into a member function.
With `ReEncodeCommitMessage` as argument, the (unclear to me) comment is not needed I guess.

Reply:
Let`s put the comment aside for now.

Key question here is: why some static method with only one usage, that would stay simple by becoming non-static, shouldn`t become non-static? Why?

Reply:
It makes it explicit that `ReEncodeCommitMessage` is used and that `ProcessDiffNotes` is stateless and does not use other members of CommitDataManager.
Then there is no need for `var module = GetModule()`.